frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx

```diff
@@ -112,77 +111,145 @@ const DefaultSourceKebab: React.FC<DefaultSourceKebabProps> = ({
   return <Kebab options={options} />;
 };
 
+const getOperatorCount = (
+  catalogSource: CatalogSourceKind,
+  packageManifests: PackageManifestKind[],
+): number =>
+  packageManifests.filter(
+    (p) =>
+      p.status?.catalogSource === catalogSource.metadata.name &&
+      p.status?.catalogSourceNamespace === catalogSource.metadata.namespace,
+  ).length;
+
+const getEndpoint = (catalogSource: CatalogSourceKind): React.ReactNode => {
+  if (catalogSource.spec.configmap) {
+    return (
+      <ResourceLink
+        kind={referenceForModel(ConfigMapModel)}
+        name={catalogSource.spec.configmap}
+        namespace={catalogSource.metadata.namespace}
+      />
+    );
+  }
+  return catalogSource.spec.image || catalogSource.spec.address;
+};
+
 export const CatalogSourceDetails: React.FC<CatalogSourceDetailsProps> = ({
-  obj,
+  obj: catalogSource,
   packageManifests,
-  subscriptions,
-  operatorGroups,
 }) => {
-  const toData = <T extends K8sResourceKind>(data: T[]) => ({ loaded: true, data });
   const { t } = useTranslation();
 
-  return !_.isEmpty(obj) ? (
-    <div className="co-catalog-details co-m-pane">
-      <div className="co-m-pane__body">
-        <div className="col-xs-4">
-          <dl className="co-m-pane__details">
-            <dt>{t('catalog-source~Name')}</dt>
-            <dd data-test-id="catalog-source-name">{obj.spec.displayName}</dd>
-          </dl>
+  const operatorCount = getOperatorCount(catalogSource, packageManifests);
+
+  const catsrcNamespace =
+    catalogSource.metadata.namespace === DEFAULT_SOURCE_NAMESPACE
+      ? 'Cluster wide'
+      : catalogSource.metadata.namespace;
+
+  return !_.isEmpty(catalogSource) ? (
+    <div className="co-m-pane__body">
+      <SectionHeading
+        text={t('catalog-source~CatalogSource details', {
+          resource: CatalogSourceModel.label,
+        })}
+      />
+      <div className="row">
+        <div className="col-sm-6 col-xs-12">
+          <ResourceSummary resource={catalogSource} />
         </div>
-        <div className="col-xs-4">
-          <dl className="co-m-pane__details">
-            <dt>{t('catalog-source~Publisher')}</dt>
-            <dd data-test-id="catalog-source-publisher">{obj.spec.publisher}</dd>
-          </dl>
+        <div className="col-sm-6 col-xs-12">
+          <div className="co-m-pane__body">
+            <DetailsItem
+              editAsGroup
+              label="Status"
+              obj={catalogSource}
+              path="status.connectionState.lastObservedState"
+            />
+            <DetailsItem label="Display Name" obj={catalogSource} path="spec.displayName" />
```

Comment:
nit: These should be sentence case "Display name," although we can change that when we localize.

Reply:
Since it would be done as part of i18n, I kept this as-is for now